ui/app/components/transit-key-actions.js

```diff
@@ -35,213 +50,142 @@ const TRANSIT_PARAMS = {
-      backend,
-      id,
-    };
-  },
-
-  checkAction() {
-    const currentAction = this.selectedAction;
-    const oldAction = this.oldSelectedAction;
-
-    this.resetParams(oldAction, currentAction);
-    set(this, 'oldSelectedAction', currentAction);
-  },
-
-  resetParams(oldAction, action) {
-    const params = copy(TRANSIT_PARAMS);
-    let paramsToKeep;
-    const clearWithoutCheck =
-      !oldAction ||
-      // don't save values from datakey
-      oldAction === 'datakey' ||
-      // can rewrap signatures — using that as a ciphertext later would be problematic
-      (oldAction === 'rewrap' && !this.key.supportsEncryption);
-
-    if (!clearWithoutCheck && action) {
-      paramsToKeep = PARAMS_FOR_ACTION[action];
-    }
+export default class TransitKeyActions extends Component {
+  @service store;
+  @service flashMessages;
+  @service router;
 
-    if (paramsToKeep) {
-      paramsToKeep.forEach((param) => delete params[param]);
-    }
-    //resets params still left in the object to defaults
-    this.clearErrors();
-    this.setProperties(params);
-    if (action === 'export') {
-      this.setExportKeyDefaults();
-    }
-  },
+  @tracked isModalActive = false;
+  @tracked errors = null;
+  @tracked props = { ...STARTING_TRANSIT_PROPS }; // Shallow copy of the object. We don't want to mutate the original.
 
-  handleError(e) {
-    this.set('errors', e.errors);
-  },
+  constructor() {
+    super(...arguments);
+    assert('@key is required for TransitKeyActions components', this.args.key);
 
-  clearErrors() {
-    this.set('errors', null);
-  },
+    if (this.firstSupportedAction === 'export' || this.args.selectedAction === 'export') {
+      this.props.exportKeyType = this.args.key.exportKeyTypes[0];
+      this.props.exportKeyVersion = this.args.key.validKeyVersions.lastObject;
+    }
+  }
+
+  get keyIsRSA() {
+    const { type } = this.args.key;
+    return type === 'rsa-2048' || type === 'rsa-3072' || type === 'rsa-4096';
+  }
 
-  triggerSuccessMessage(action) {
-    const message = SUCCESS_MESSAGE_FOR_ACTION[action];
-    if (!message) return;
-    this.flashMessages.success(message);
-  },
+  get firstSupportedAction() {
+    return this.args.key.supportedActions[0];
+  }
 
   handleSuccess(resp, options, action) {
-    let props = {};
     if (resp && resp.data) {
       if (action === 'export' && resp.data.keys) {
         const { keys, type, name } = resp.data;
         resp.data.keys = { keys, type, name };
       }
-      props = assign({}, props, resp.data);
+      this.props = { ...this.props, ...resp.data };
     }
     if (options.wrapTTL) {
-      props = assign({}, props, { wrappedToken: resp.wrap_info.token });
+      this.props = { ...this.props, ...{ wrappedToken: resp.wrap_info.token } };
     }
-    if (!this.isDestroyed && !this.isDestroying) {
-      this.toggleProperty('isModalActive');
-      this.setProperties(props);
+    // open the modal
+    this.isModalActive = !this.isModalActive;
```

Comment:
Can we set this to true instead of `!this.isModalActive`, then we could remove the comment and not worry about getting into some where state where the toggle is borked